cmd/opera/launcher/validator.go

```diff
@@ -32,25 +32,29 @@ var validatorPasswordFlag = cli.StringFlag{
 // setValidatorID retrieves the validator ID either from the directly specified
 // command line flags or from the keystore if CLI indexed.
 func setValidator(ctx *cli.Context, cfg *emitter.Config) error {
+	if ctx.GlobalIsSet(validatorIDFlag.Name) && ctx.GlobalIsSet(FakeNetFlag.Name) && cfg.Validator.ID != 0 {
```

Comment:
This check has to be after `if ctx.GlobalIsSet(FakeNetFlag.Name) {`, or better inside that `if block`